taskgraph/Task.py

```diff
@@ -628,11 +628,11 @@ def add_task(
             hardlink_allowed (bool): if ``copy_duplicate_artifact`` is True,
                 this will allow a hardlink rather than a copy when needed.
             transient_run (bool): if True a call with an identical execution
-                hash will be reexecuted on a subsequent instantiation of a
-                future TaskGraph object. If a duplicate task is submitted
+                hash will be reexecuted on the same or subsequent instantiation
+                of a TaskGraph object. If a duplicate task is submitted
                 to the same object it will not be re-run in any scenario.

```

Comment:
Great point! And thanks for pointing this out because it helps me get it straight in my head. Abstractly, Taskgraph is a dependency graph and an instance of a TaskGraph object traverses that dependency graph executing Tasks as allowed and needed. My intent for `transient_run` is that every time a Task is iterated over in the dependency graph (the equivalent of an instantiation of TaskGraph), that Task be executed. 

The confusing stuff about duplicate Tasks may be me over engineering TaskGraph. Since TaskGraph's abstract model is a graph, it's the equivalent of a set, it doesn't make sense to have duplicate nodes because it's the same node. In practice you'd have a similar confusion -- why indicate that a Task exists twice? So TaskGraph does detect this issue and rather than raise an error it logs a warning and discards the Task. I added this because I was anticipating, and use, cases where I'm not sure if a Task has been added to the graph yet and TaskGraph otherwise has no way of querying its state.

SO that's all coming down to this mishmash of docstring that I wrote. I will rephrase it to follow the abstraction here.

Reply:
Okay I updated that docstring in 34416c438141c99d9ad93db7f58fa4cd1f36858c